rfcs/20210510-sum-with-zero-point/README.md

```diff
@@ -0,0 +1,158 @@
+                dnnl_post_ops_t post_ops, float scale, dnnl_data_type_t data_type,
+                int32_t zero_point);
+
+        // include/oneapi/dnnl/dnnl.hpp
+        void append_sum(float scale = 1.f,
+                 memory::data_type data_type = memory::data_type::undef,
+                 int32_t zero_point = 0) {
+            if (data_type == memory::data_type::undef)
+                error::wrap_c_api(dnnl_post_ops_append_sum(get(), scale),
+                        "could not append a sum post-op");
+            else
+                if (zero_point == 0)
+                    error::wrap_c_api(dnnl_post_ops_append_sum_v2(
+                                              get(), scale,
+                                              memory::convert_to_c(data_type)),
+                            "could not append a sum post-op");
+                else
+                    error::wrap_c_api(dnnl_post_ops_append_sum_v3(
+                                              get(), scale,
+                                              memory::convert_to_c(data_type)),
+                                              zero_point,
+                            "could not append a sum post-op");
+        }
+        ```
+    - Pros:
+        - It is backwards compatible.
+        - Allows to use zero point without scale (scale == 1.0).
+    - Cons:
+        - It is not the simplest solution
+    - Example:
+        ```cpp
+        // Zero point is the last parameter to keep existing code working.
+        attr.append_sum(scale, dt);
+        attr.append_sum(scale, dt, zp);
+        ```
+
+2. Extend existing `append_sum(scale, dt)` and `dnnl_post_ops_append_sum_v2` to
+support zero point.
+    - API:
+        ```cpp
+        // include/oneapi/dnnl/dnnl.h
+        dnnl_status_t DNNL_API dnnl_post_ops_append_sum_v2(
+                dnnl_post_ops_t post_ops, float scale, int32_t zero_point,
+                dnnl_data_type_t data_type);
+
+        // include/oneapi/dnnl/dnnl.hpp
+            void append_sum(float scale = 1.f, int32_t zero_point = 0,
+                    memory::data_type data_type = memory::data_type::undef) {
+                if (data_type == memory::data_type::undef)
+                    error::wrap_c_api(dnnl_post_ops_append_sum(get(), scale),
+                            "could not append a sum post-op");
+                else
+                    error::wrap_c_api(dnnl_post_ops_append_sum_v2(
+                                              get(), scale, zero_point,
+                                              memory::convert_to_c(data_type)),
+                            "could not append a sum post-op");
+            }
+        ```
+    - Pros:
+        - Very simple solution.
+        - Universal solution.
+    - Cons:
+        - It is not backwards compatible for C API.
+    - Example:
+        ```cpp
+        // Zero point is the last parameter to keep existing code working.
+        attr.append_sum(scale, dt);
+        ```
```

Comment:
Option 2 is a non starter if we want this feature to appears in v2.x, as we don't break API/ABI between minor releases.

Reply:
Therefore it is not preferred. Do You want to remove them?

Reply:
I think yes we can remove it. And we will rename the `_v*` functions anyway with next major release.

Reply:
Removed.